packages/graphback-codegen-schema/src/plugin/SchemaCRUDPlugin.ts

```diff
@@ -0,0 +1,234 @@
+import * as pluralize from "pluralize";
+import { gqlSchemaFormatter, jsSchemaFormatter, tsSchemaFormatter } from '../writer/schemaFormatters';
+import { printSortedSchema } from '../writer/schemaPrinter';
+
+/**
+ * Configuration for Schema generator CRUD plugin
+ */
+export interface SchemaCRUDPluginConfig {
+    // output format for schema string
+    format: 'ts' | 'js' | 'gql'
+}
+
+export const SCHEMA_CRUD_PLUGIN_NAME = "SchemaCRUD";
+
+/**
+ * Graphback CRUD operations plugin
+ * 
+ * Plugins adds additional Queries, Mutations and Subscriptions into the Schema along 
+ * with required input types and scalars. Plugin can be used automatically define best 
+ * patterns for CRUD operations on top of GraphQL Schema
+ * Plugin checkes all types annotated with model
+ * 
+ * Used graphql metadata:
+ * 
+ * - model: marks type to be processed by CRUD generator
+ * - crud: controls what types of operations can be generated. 
+ * For example crud.update: false will disable updates for type
+ */
+export class SchemaCRUDPlugin extends GraphbackPlugin {
+
+    private pluginConfig: SchemaCRUDPluginConfig;
+
+    constructor(pluginConfig?: SchemaCRUDPluginConfig) {
+        super()
+        this.pluginConfig = Object.assign({ format: 'gql' }, pluginConfig);
+
+    }
+
+    public transformSchema(metadata: GraphbackCoreMetadata): GraphQLSchema {
+        const schema = metadata.getSchema()
+        const models = metadata.getModelDefinitions();
+        if (models.length === 0) {
+            this.logWarning("Provided schema has no models. Returning original schema without any changes.")
+
+            return schema;
+        }
+
+        const modelsSchema = this.buildSchemaForModels(models);
+
+        return mergeSchemas({ schemas: [modelsSchema, schema] });
+    }
+
+    /**
+     * Create resolvers function that 
+     * @param inputContext 
+     * @param options 
+     */
+    public transformSchemaToString(schema: GraphQLSchema) {
+        const schemaString = printSortedSchema(schema);
+        if (this.pluginConfig) {
+            if (this.pluginConfig.format === 'ts') {
+                return tsSchemaFormatter.format(schemaString)
+            }
+            if (this.pluginConfig.format === 'js') {
+                return jsSchemaFormatter.format(schemaString)
+            }
+            if (this.pluginConfig.format === 'gql') {
+                return gqlSchemaFormatter.format(schemaString)
+            }
+        }
+        throw Error("Invalid format specified. `options.format` supports only `ts`, `js` and `gql` flags");
+    }
+
+
+
+    public getPluginName() {
+        return SCHEMA_CRUD_PLUGIN_NAME;
+    }
+
+    private buildSchemaForModels(models: ModelDefinition[]) {
+        let queryTypes = {};
+        let mutationTypes = {};
+        let subscriptionTypes = {};
+
+        for (const model of Object.values(models)) {
+            if (model.crudOptions.disableGen) {
```

Comment:
Missing the check for global CRUD config. is this something I should add or did you plan to add it in this plugin?